testonly/hammer/hammer.go

```diff
@@ -491,37 +491,30 @@ func (s *hammerState) retryOneOp(ctx context.Context) (err error) {
 	return firstErr
 }
 
-func (s *hammerState) performOp(ctx context.Context, ep MapEntrypointName, prng *rand.Rand) error {
-	switch ep {
-	case GetLeavesName:
-		return s.validReadOps.getLeaves(ctx, prng)
-	case GetLeavesRevName:
-		return s.validReadOps.getLeavesRev(ctx, prng)
-	case GetSMRName:
-		return s.validReadOps.getSMR(ctx, prng)
-	case GetSMRRevName:
-		return s.validReadOps.getSMRRev(ctx, prng)
-	case SetLeavesName:
-		return s.setLeaves(ctx, prng)
-	default:
-		return fmt.Errorf("internal error: unknown entrypoint %s selected for valid request", ep)
-	}
+type readOps interface {
+	getLeaves(context.Context, *rand.Rand) error
```

Comment:
I wonder if `rand.Rand` should be baked into the implementation of the type rather than the interface. I'm guessing that the caller uses the same instance of `Rand` for all these calls anyway. The fact of re-creating it before each call complicates this a bit maybe...

Reply:
I initially thought this too, but in the event of a retry being required it actually passes in a prng that returns the same sequence as before for predictability during the retry. This needs to stay as a parameter for now. See `retryOneOp` implementation for where this prng regeneration is performed.